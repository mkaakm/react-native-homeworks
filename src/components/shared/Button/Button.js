import React from "react";
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

export default function Button({text, type = 'light', style, onPress}) {
    const className = type === 'light' ? styles.lightButton : styles.darkButton

    return (
        <View style={[className, style]}>
            <TouchableOpacity onPress={onPress}>
                <Text style={styles.buttonText}>{text}</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    lightButton: {
        color: "#000",
        borderColor: '#fff',
        borderRadius: 25,
        backgroundColor: '#fff',
        textAlign:'center',
        borderStyle:'solid',
        borderWidth: 1,
        width: 100,
        paddingTop: 10,
        paddingBottom: 10,
    },
    darkButton:{
        color:'#fff',
        borderColor: 'red',
        borderRadius: 25,
        backgroundColor: 'red',
        textAlign:'center',
        borderStyle:'solid',
        borderWidth: 1,
        width: 100,
        paddingTop: 10,
        paddingBottom: 10,
    },
    buttonText:{
        textAlign:'center',
    }
});
