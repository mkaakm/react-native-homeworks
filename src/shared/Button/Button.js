import React from "react";
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import {COLORS} from "../../color";

export default function Button({text, type = 'light', style, onPress}) {
    // const className = (type === 'light') ? styles.lightButton : styles.darkButton

    return (
        <View style={styles.lightButton}>
            <TouchableOpacity onPress={onPress}>
                <Text style={styles.buttonText}>{text}</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    lightButton: {
        color: "#000",
        borderColor: COLORS.primary,
        borderRadius: 25,
        backgroundColor: COLORS.primary,
        textAlign:'center',
        borderStyle:'solid',
        borderWidth: 1,
        width: 100,
        paddingVertical:10
    },
    darkButton:{
        color:COLORS.primary,
        borderColor: COLORS.secondary,
        borderRadius: 25,
        backgroundColor: COLORS.secondary,
        textAlign:'center',
        borderStyle:'solid',
        borderWidth: 1,
        width: 100,
        paddingTop: 10,
        paddingBottom: 10,
    },
    buttonText:{
        textAlign:'center',
    },

});
