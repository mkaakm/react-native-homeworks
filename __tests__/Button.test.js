import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/shared/Button'
import {StyleSheet} from 'react-native';
import { fireEvent, render } from '@testing-library/react-native';

test('renders correctly', () => {
    const tree = renderer.create(<Button text='button' style={styles.button}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('onPress button correctly', () => {
    const mockedHandler = jest.fn();
    const {getByText} = render(<Button onPress={mockedHandler} text='button'/>);

    fireEvent.press(getByText('button'), 'fire');
    expect(mockedHandler).toHaveBeenCalledWith('fire');
});

const styles = StyleSheet.create({
    button:{
        textAlign:"center"
    }
})
