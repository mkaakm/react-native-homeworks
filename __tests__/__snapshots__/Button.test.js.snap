// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#fff",
        "borderRadius": 25,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "#000",
        "paddingVertical": 10,
        "textAlign": "center",
        "width": 100,
      },
      Object {
        "textAlign": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      button
    </Text>
  </View>
</View>
`;
